### PR TITLE
Allow component-level provider overrides

### PR DESCRIPTION
**Pull request by yjaaidi**

### CODE REVIEW

File: projects/angular-dev-server/src/lib/getWebpackConfig.ts [hunk 1]
```diff
@@ -1,48 +1,59 @@
...
+          // Angular linker needed to link partial-ivy code
+          // See https://angular.io/guide/creating-libraries#consuming-partial-ivy-code-outside-the-angular-cli
+          test: /\.m?js$/,
+          use: {
+            loader: 'babel-loader',
+            options: {
+              plugins: [
+                [
+                  '@angular/compiler-cli/linker/babel',
+                  { linkerJitMode: jitMode },
```
Comment (yjaaidi) [inline projects/angular-dev-server/src/lib/getWebpackConfig.ts:39]: This was the missing magic!

File: projects/angular/src/lib/mount.ts [hunk 2]
```diff
@@ -85,11 +116,9 @@ export function mount<T extends object>(
 
   const fixture = setupFixture(component, testBed, autoDetectChanges);
 
-  testBed.compileComponents();
```
Comment (yjaaidi) [inline projects/angular/src/lib/mount.ts:88]: This shouldn't be needed as it's implicitly done by TestBed anyway.
The only async thing is fetching resources (styles/templates) but this is done at build time... so it doesn't make much sense today.

File: projects/angular-dev-server/src/lib/getWebpackConfig.ts (hunk 1 above)
Comment (jordanpowell88) [reply projects/angular-dev-server/src/lib/getWebpackConfig.ts:39]: Makes total sense too. Thank you so much for discovering this little gem!

File: projects/angular-dev-server/src/lib/getWebpackConfig.ts [hunk 3]
```diff
@@ -1,48 +1,59 @@
-import { AngularWebpackPlugin } from "@ngtools/webpack";
-import { Configuration } from "webpack";
+import { AngularWebpackPlugin } from '@ngtools/webpack';
+import { Configuration } from 'webpack';
 
 export async function getWebpackConfig(tmpDir: string): Promise<Configuration> {
+  /* We are using JIT mode as AOT has issues with overriding components (e.g. component providers etc...)
+   * Cf. https://github.com/jordanpowell88/angular-ct/issues/36 */
+  const jitMode = true;
```
Comment (jordanpowell88) [inline projects/angular-dev-server/src/lib/getWebpackConfig.ts:7]: We need to pass this in as an parameter of this `getWebpackConfig() defaulted to true

Comment (yjaaidi) [reply projects/angular-dev-server/src/lib/getWebpackConfig.ts:7]: +1 but we've got to fix AOT first as it produces the following error:
```
class doesn't have @Component decorator or is missing metadata
```

we should probably wait to fix this first.

File: projects/angular-dev-server/src/lib/getWebpackConfig.ts (hunk 1 above)
Comment (yjaaidi) [reply projects/angular-dev-server/src/lib/getWebpackConfig.ts:39]: Actually, it also works if we disable the linker but it's less efficient as the linker does some caching and stuff... and it's more aligned to the CLI.

File: projects/angular-dev-server/src/lib/getWebpackConfig.ts (hunk 3 above)
Comment (jordanpowell88) [reply projects/angular-dev-server/src/lib/getWebpackConfig.ts:7]: Works for me. We'll just need to create a follow up Issue

**Pull request merged**